src/v/utils/uuid.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "bytes/details/out_of_range.h"
+#include "reflection/adl.h"
+
+#include <absl/hash/hash.h>
+#include <boost/functional/hash.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <vector>
+
+// Wrapper around Boost's UUID type suitable for serialization with serde.
+// Provides utilities to construct and convert to other types.
+//
+// Expected usage is to supply a UUID v4 (i.e. based on random numbers). As
+// such, serialization of this type simply serializes 16 bytes.
+struct uuid_t {
+public:
+    static constexpr auto length = 16;
+    using underlying_t = boost::uuids::uuid;
+
+    underlying_t uuid;
```

Comment:
IMO i'd make this class immutable, making this field private, provide a const accessor (like operator() or *) and then having a static method like `static uuid create()` that returns a random UUID, properly initialized, constructed, etc.

Reply:
Hmm, not a bad idea to have a static create. Maybe I'll add that in a follow up (though working on some other stuff first unless you feel strongly about this). Probably better than leaking `boost/uuids/random_generator` everywhere.

Not sure about making it immutable though -- not sure whether it'd play well with serde.